src/WebSdk/Publish/Tasks/Tasks/TransformWebConfig.cs

```diff
@@ -121,16 +130,27 @@ public override bool Execute()
             }
 
             string outputFile = Path.GetFileName(TargetPath);
-            XDocument transformedConfig = WebConfigTransform.Transform(
-                webConfigXml,
-                outputFile,
-                IsAzure,
-                UseAppHost,
-                ExecutableExtension,
-                AspNetCoreModuleName,
-                AspNetCoreHostingModel,
-                EnvironmentName,
-                ProjectFullPath);
+
+            // For ASP.NET Framework web applications do not add any thing mentioning ASP.NET Core in it's Web.config file.
+            // It could break it at runtime when it tries to use the ASP.NET Core Module at runtime with it's version of ASP.NET
+            // (and as such the other ASP.NET Module for .NET Framework).
+            // Also it's because I expect the files between the web.config on non-SDK style and SDK style to be exact copies
+            // provided that it still targets .NET Framework.
+            // This will capture all apps targeting .NET Framework and skip adding the 2 aspNetCore elements to the web.config
+            // but it is worth it to not have breakage at all in web.config.
+            // Fixes: https://github.com/dotnet/sdk/issues/34595
+            XDocument transformedConfig = TargetFrameworkIdentifier != ".NETFramework" ||
```

Comment:
This check is also incorrect because ASP.NET still supports running ASP.NET Core on full framework. Adding this check will potentially cause many valid scenarios to fail.